From f88fe33b8c28d732b624c521a9a50a7e661606ed Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 11 Jun 2012 15:22:59 +0900
Subject: [PATCH 32/97] h264parse: set GST_BUFFER_FLAG_PREROLL flag if buffer
 contains SPS or PPS

This implemantation notifies the next plugin that the buffer
containing SPS or PPS.
---
 gst/h264parse/gsth264parse.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/gst/h264parse/gsth264parse.c b/gst/h264parse/gsth264parse.c
index d39eb35..f1c0e62 100644
--- a/gst/h264parse/gsth264parse.c
+++ b/gst/h264parse/gsth264parse.c
@@ -1689,6 +1689,7 @@ gst_h264_parse_push_buffer (GstH264Parse * h264parse, GstBuffer * buf)
           GST_BUFFER_SIZE (nals->data));
       GST_BUFFER_TIMESTAMP (nals->data) = GST_BUFFER_TIMESTAMP (buf);
       GST_BUFFER_DURATION (nals->data) = 0;
+      GST_BUFFER_FLAG_SET (nals->data, GST_BUFFER_FLAG_PREROLL);
 
       gst_buffer_set_caps (nals->data, h264parse->src_caps);
       (void) gst_pad_push (h264parse->srcpad, nals->data);
-- 
1.7.10.4

